airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -443,6 +443,10 @@ def execute(self, context: Context) -> None:  # type: ignore[override]
                     method_name="execute_complete",
                 )
             self._handle_job_error(job)
+            # job.result() returns a RowIterator. Mypy expects an instance of SupportsNext[Any] for
+            # the next() call which the RowIterator does not resemble to. Hence, ignore the arg-type error.
+            records = next(job.result())  # type: ignore[arg-type]
+            self.check_value(records)
```

Comment:
If we're checking the values here, can we remove the `check_value` call in the `BigQueryValueCheckTrigger`?

https://github.com/apache/airflow/blob/81b85ebcbd241e1909793d7480aabc81777b225c/airflow/providers/google/cloud/triggers/bigquery.py#L441

I'm not really sure why the check_value there wasn't picking up the failed status 🤔 I'd assume that if the check failed it would raise an exception which would then be caught and returned as a failed trigger status:

https://github.com/apache/airflow/blob/81b85ebcbd241e1909793d7480aabc81777b225c/airflow/providers/google/cloud/triggers/bigquery.py#L451-L454

Reply:
I gave the DAG a local run and saw that the task was not getting deferred at all, which means it does not defer.

This happens because of https://github.com/apache/airflow/blob/3ae6b4e86fe807c00bd736c59df58733df2b9bf9/airflow/providers/google/cloud/operators/bigquery.py#L307 
The running state of the job evaluates to False and it does not defer. Had it deferred, the trigger would check it well and pass on the execution to execute_complete in the operator.

But in scenarios where it does not defer, I have added this check later to actually achive what the operator is meant to do :) 

Sorry I missed explaining this earlier, hope it makes sense now.

Reply:
Oh interesting, nice catch!